lib/puppet/parser/functions/remove_duplicate_whitespace.rb

```diff
@@ -0,0 +1,35 @@
+#
+#  remove_duplicate_whitespace.rb
+#
+module Puppet::Parser::Functions
+  newfunction(:remove_duplicate_whitespace, :type => :rvalue, :doc => <<-DOC
+    This function replaces one or multiple whitespaces with a single one from a string or from
+    every string inside an array.
+
+    *Examples:*
+
+        strip("    aaa   bla")
+
+    Would result in: " aaa bla"
+    DOC
+             ) do |arguments|
+
+    raise(Puppet::ParseError, "remove_duplicate_whitespace(): Wrong number of arguments given (#{arguments.size} for 1)") if arguments.empty?
+
+    value = arguments[0]
+
+    unless value.is_a?(Array) || value.is_a?(String)
+      raise(Puppet::ParseError, 'remove_duplicate_whitespace(): Requires either array or string to work with')
+    end
+
+    result = if value.is_a?(Array)
+               value.map { |i| i.is_a?(String) ? i.gsub(/\s+/, ' ') : i }
```

Comment:
How about keeping the original whitespace char. Something like ```gsub(/(\s)\s+/, '\1')```

Reply:
Would that make a huge difference?

Reply:
I'm wondering if people like using tabs or perhaps tabs are needed somehow?